test/jbuild

```diff
@@ -1,42 +1,14 @@
-(* -*- tuareg -*- *)
-#require "unix"
-
-let flags = function
-  | [] -> ""
-  | pkgs ->
-    let cmd = "ocamlfind ocamlc -verbose" ^ (
-        List.fold_left (fun acc pkg -> acc ^ " -package " ^ pkg) "" pkgs
-      ) in
-    let ic = Unix.open_process_in
-        (cmd ^ " | grep -oEe '-ppx (\"([^\"\\]|\\.)+\"|\\w+)'")
-    in
-    let rec go ic acc =
-      try go ic (acc ^ " " ^ input_line ic) with End_of_file -> close_in ic; acc
-    in
-    go ic ""
-
-let flags = flags ["ppx_deriving_rpc"; "ppx_sexp_conv"]
-
-let coverage_rewriter =
-  let is_coverage = try Unix.getenv "BISECT_ENABLE" = "YES" with Not_found -> false in
-  if is_coverage then
-    "(preprocess (pps (bisect_ppx -conditional)))"
-  else
-    ""
-
-let () = Printf.ksprintf Jbuild_plugin.V1.send {|
 (jbuild_version 1)
 
 (executable (
   (name xenopsd_test)
   (libraries (oUnit xenopsd))
-  (flags (:standard -w -39 %s))
-  %s
+  (flags (:standard -w -39))
+  (preprocess (pps (ppx_deriving_rpc ppx_sexp_conv bisect_ppx -conditional)))
```

Comment:
This works for xs-opam, but requires bisect_ppx to be present and build time and trust it not to modify your code when the env var is not present.
I think this might fail in Jenkins where we do not install bisect-ppx, to prevent it accidentally enabling coverage profiling in production code.

See here on how to do this conditionally: https://github.com/xapi-project/message-switch/blob/51955847b3dd52f68f4ea292acc018c885e6593e/cli/dune

Reply:
Although if we decide that we trust this code https://github.com/aantron/bisect_ppx/blob/1114b91b04307cbd2dd040bc992bea5bc2a310cb/src/ppx/register.ml then we could move bisect-ppx from upstream-extra to upstream in xs-opam and use the simplified form you have here.

Reply:
I looked at the code of bisect_ppx and ocaml-migrate-parsetree to see what code is executed when `BISECT_ENABLE` is not `YES`. It is simple and executes `Ast_405.shallow_identity` on the AST, which is an identity operation. I would trust it and suggest we carry bisect_ppx along.

Reply:
Ok, we'll need a new xs-opam release for that

Reply:
Actually, we don't. `bisect_ppx` is part of our `upstream` libraries already.